modules/tinymce/src/core/main/ts/textpatterns/core/BlockPattern.ts

```diff
@@ -73,8 +74,21 @@ const findPatterns = (editor: Editor, patterns: BlockPattern[]): BlockPatternMat
     // Get the block text
     const blockText = block.textContent ?? '';
 
-    // Find the pattern
-    const matchedPattern = findPattern(patterns, blockText);
+    // TINY-8781: TODO: text_patterns should announce their changes for accessibility
+    const extraPatterns = patternSet.dynamicPatternsLookup({
+      text: blockText,
+      block,
+      // Block patterns do not allow trailing spaces currently. This is related to TINY-8779
+      allowTrailingSpaces: false
+    });
+    // search in the dynamic patterns first
+    const patterns = getBlockPatterns(extraPatterns);
+    const matchedPattern = findPattern(patterns, blockText).orThunk(() => {
+      // Search in the static patterns
+      const patterns = getBlockPatterns(patternSet.blockPatterns);
+      return findPattern(patterns, blockText);
+    });
```

Comment:
I've been a little concerned about the performance implications of textpattern for a while, but this is looking more and more expensive.

Can we please log a technical debt task under either this epic or the small changes area to improve the way this operates?

Dynamic patterns will always be more expensive, that's expected, but I see a lot of logic that could be run in response to option changes instead of re-loaded and re-analysed every time a trigger key is pressed.

Reply:
@TheSpyder , I created TINY-8904 as per your request. Can you please have a look as I think additional info would be helpful.